CHANGELOG.md

```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Mode errors are better explained and include a fix suggestion (#619)
 - Typechecking now reports errors when instance overrides are not compatible with
   the original definition (#622)
+- the parser and tests are using the id generator now (#647)
```

Comment:
Is this a user-facing breaking change? Seems like just an internal optimization?

Reply:
I hope it's not breaking anything, but if something breaks, we could track it by looking at the changelog, or the commit log. I am not sure.

Reply:
If we are following “keep a change log”, changes we shouldn’t be breaking and don’t change the api at all aren’t really meant for the changelog I think?

> The purpose of a commit is to document a step in the evolution of the source code. Some projects clean up commits, some don't.
> 
> The purpose of a changelog entry is to document the noteworthy difference, often across multiple commits, to communicate them clearly to end users.

https://keepachangelog.com/en/1.0.0/

so this seems pretty clearly like a commit thing to me. Let’s be sure we don’t break things, by adding new tests if needed? But i worry adding stuff as breaking changes because it *might* break something in some unforeseen  way just makes that part of the changelog unreliable and noisy.

End users don’t need to know about refactoring :)

Reply:
sure, let's remove it then

Reply:
removed